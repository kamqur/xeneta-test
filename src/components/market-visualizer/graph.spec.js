import React from "react";
import { shallow } from "enzyme";
import ReactEcharts from "echarts-for-react";

import Graph from "./graph";

describe("Testing Graph Componenet", () => {
  it("snapshort testing", () => {
    const component = shallow(
      <Graph
        data={[
          {
            day: "day",
            mean: 1,
            low: 1,
            high: 1
          }
        ]}
        errors={[]}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("data is not passed", () => {
    const component = shallow(<Graph errors={[]} />);
  });

  it("there are errors", () => {
    const component = shallow(
      <Graph
        errors={{
          message: "error"
        }}
      />
    );
    const errors = component.find(".errors");
    expect(errors).toHaveLength(1);
  });

  it("Chart will be rendered", () => {
    const component = shallow(
      <Graph
        data={[
          {
            day: "day1",
            mean: 1,
            low: 1,
            high: 1
          },
          {
            day: "day2",
            mean: 1,
            low: 1,
            high: 1
          }
        ]}
        errors={[]}
      />
    );

    const chart = component.find(ReactEcharts);
    expect(chart).toHaveLength(1);
  });

  it("Chart will be rendered", () => {
    const component = shallow(
      <Graph
        data={[
          {
            day: "day1",
            mean: 1,
            low: 1,
            high: 1
          },
          {
            day: "day2",
            mean: 1,
            low: 1,
            high: 1
          }
        ]}
        errors={[]}
      />
    );

    const chart = component.find(ReactEcharts);
    expect(chart).toHaveLength(1);
  });

  it("Chart will be rendered", () => {
    const component = shallow(
      <Graph
        data={[
          {
            day: "day1",
            mean: 1,
            low: 1,
            high: 1
          },
          {
            day: "day1",
            mean: 1,
            low: 1,
            high: 1
          }
        ]}
        errors={[]}
      />
    );

    const chart = component.find(ReactEcharts);
    expect(chart).toHaveLength(1);
  });
});
