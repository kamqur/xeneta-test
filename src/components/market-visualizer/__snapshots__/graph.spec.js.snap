// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Graph Componenet snapshort testing 1`] = `
<div
  style={
    Object {
      "height": "500px",
      "paddingTop": "5%",
      "width": "100%",
    }
  }
>
  <EChartsReact
    option={
      Object {
        "legend": Object {
          "data": Array [
            "Mean",
            "Low",
            "High",
          ],
        },
        "responsive": true,
        "scale": true,
        "series": Array [
          Object {
            "data": Array [
              1,
            ],
            "name": "Mean",
            "smooth": true,
            "type": "line",
          },
          Object {
            "data": Array [
              1,
            ],
            "name": "Low",
            "smooth": true,
            "type": "line",
          },
          Object {
            "data": Array [
              1,
            ],
            "name": "High",
            "smooth": true,
            "type": "line",
          },
        ],
        "tooltip": Object {
          "trigger": "axis",
        },
        "xAxis": Object {
          "data": Array [
            "day",
          ],
          "type": "category",
        },
        "yAxis": Object {
          "type": "value",
        },
      }
    }
  />
</div>
`;
